src/components/SingleBlock/SingleBlockClusterGraph.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ClusterGraph from "../Graphs/ClusterGraph";
+
+interface Props {
+    data: any;
+}
+
+export default function SingleBlockClusterGraph({ data }: Props) {
+
+    const singleBlockDataArray: any[] = [];
+
+    for (let i in data) {
+        if (data.hasOwnProperty(i)) {
+            let items = data[i];
+            for (let j in items) {
+                // inputs, kernels, outputs
+                if (data[i].hasOwnProperty(j)) {
+                    let children = data[i][j];
+                    for (let k in children) {
+                        // inputs[0], inputs[1] etc
+                        if (data[i][j].hasOwnProperty(k)) {
+                            let child = data[i][j][k];
+                            child.group = j;
+                            child.size = data[i][j].length;
+                            child.color = j === 'inputs' ? '#F97C0D' : j === 'kernels' ? '#FB576D' : '#2274AF';
+                            singleBlockDataArray.push(child);
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
What's going on here?

Reply:
It's going through the data to get the inputs, kernels and outputs and pushing it into a better array for using in the graph. Also adding in the size, group and color in the new array.

Reply:
It would be better to get the kernels, inputs, outputs explicitly before passing the data into the clustergraph.
```javascript
const {inputs, outputs, kernels}= block.body
inputs.forEach(i => {
  i.group = 'inputs'
  i.color = '#123123'
  i.size = size
})

// Repeat for outputs kernels
const data = [inputs, outputs, kernels]
 render() {<SingleBlockClusterGraph data={data}/>}
```